### PR TITLE
Issue #56: Extract strings for translation

### PR DESCRIPTION
**Pull request by hannesgr**
## Description

    Extracted english strings in components for translation. About page not included

## Related issues

       - Contributes to https://github.com/neherlab/covid19_scenarios/issues/56

### CODE REVIEW

**Comment by ivan-aksamentov** [comment]
I merged https://github.com/neherlab/covid19_scenarios/pull/70/v first. Could you please rebase?

Do we really have to replace all spaces with dashes? I'd be glad if you say no, because our scientists would hate the idea :) 

Someone should write a small guide on how to modify the strings properly. Somewhere in the README.md for now, and I will figure where to put it later.

**Comment by hannesgr** [comment]
> @hannesgr I merged #70 first. Could you please rebase?

Done!

> Do we really have to replace all spaces with dashes? I'd be glad if you say no, because our scientists would hate the idea :)

No :) I've updated so we use spaces instead of dashes

**Comment by hannesgr** [comment]
Noticed I had missed the presets, I've forced pushed now

File: src/components/Main/Results/ResultsCard.tsx [hunk 1]
```diff
@@ -74,16 +78,14 @@ function ResultsCard({ canRun, severity, result, caseCounts }: ResutsCardProps)
...
+      help={t('This section contains simulation results')}
       defaultCollapsed={false}
     >
       <Row noGutters>
         <Col>
           <p>
-            {`This output of a mathematical model depends on model assumptions and parameter choices.
-              We have done our best (in limited time) to check the model implementation is correct.
-              Please carefully consider the parameters you choose and interpret the output with caution.`}
+            {t('Results disclaimer')}
```
Comment (xrmx) [inline src/components/Main/Results/ResultsCard.tsx]: Am not sure you really want to substitute the text to translate with a generic label

File: src/components/Main/Scenario/ScenarioCardContainment.tsx [hunk 2]
```diff
@@ -35,16 +38,16 @@ function ScenarioCardContainment({ scenarioState, errors, touched, scenarioDispa
   return (
     <CardWithDropdown
       identifier="containmentScenario"
-      label={<h5 className="p-0 d-inline text-truncate">Mitigation</h5>}
-      help="Reduction of transmission through mitigation measures over time. Different presets with variable degree of reduction can be selected from the dropdown."
+      label={<h5 className="p-0 d-inline text-truncate">{t('Mitigation')}</h5>}
+      help={t('reduction of transmission through mitigation measures over time different presets with variable degree of reduction can be selected from the dropdown')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardContainment.tsx]: I don't think you want to change the base string.

File: src/components/Main/Scenario/ScenarioCardContainment.tsx [hunk 3]
```diff
@@ -56,8 +59,7 @@ function ScenarioCardContainment({ scenarioState, errors, touched, scenarioDispa
       </div>
       <div>
         <p>
-          Drag black dots with the mouse to simulate how infection control affects the outbreak trajectory. One is no
-          infection control, zero is complete prevention of all transmission.
+          {t('Drag black dots with the mouse to simulate how infection control affects the outbreak trajectory')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardContainment.tsx]: Same.

File: src/components/Main/Results/AgeBarChart.tsx [hunk 4]
```diff
@@ -78,17 +81,17 @@ export function AgeBarChart({ data, rates }: SimProps) {
                   top: 3,
                 }}
               >
-                <XAxis dataKey="name" label={{ value: 'Age', position: 'insideBottom', offset: -3 }} />
+                <XAxis dataKey="name" label={{ value: t('Age'), position: 'insideBottom', offset: -3 }} />
                 <YAxis
                   label={{
-                    value: '% of total',
+                    value: t('percent of total'),
```
Comment (xrmx) [inline src/components/Main/Results/AgeBarChart.tsx]: You changed the translatable string

File: src/components/Main/Results/AgeBarChart.tsx [hunk 5]
```diff
@@ -78,17 +81,17 @@ export function AgeBarChart({ data, rates }: SimProps) {
...
                     angle: -90,
                     position: 'insideLeft',
                   }}
                 />
                 <CartesianGrid strokeDasharray="3 3" />
                 <Tooltip />
-                <Bar dataKey="fraction" fill="#aaaaaa" name="% of total" />
+                <Bar dataKey="fraction" fill="#aaaaaa" name={t('percent of total')} />
```
Comment (xrmx) [inline src/components/Main/Results/AgeBarChart.tsx]: same

File: src/components/Main/Results/DeterministicLinePlot.tsx [hunk 6]
```diff
@@ -107,16 +110,16 @@ export function DeterministicLinePlot({ data, userResult, logScale, caseCounts }
     }))
   const scatterToPlot: LineProps[] = []
   const linesToPlot: LineProps[] = [
-      {key:'hospitalBeds', color: colors.hospitalBeds, name:'Total hospital beds', legendType:"none"},
-      {key:'ICUbeds', color: colors.ICUbeds, name:'Total ICU/ICM beds', legendType:"none"},
-      {key:'susceptible', color: colors.susceptible, name:'Susceptible', legendType:"line"},
+      {key:'hospitalBeds', color: colors.hospitalBeds, name:t('Total hospital beds'), legendType:"none"},
+      {key:'ICUbeds', color: colors.ICUbeds, name:t('Total icu icm beds'), legendType:"none"},
```
Comment (xrmx) [inline src/components/Main/Results/DeterministicLinePlot.tsx]: same

File: src/components/Main/Results/DeterministicLinePlot.tsx [hunk 7]
```diff
@@ -107,16 +110,16 @@ export function DeterministicLinePlot({ data, userResult, logScale, caseCounts }
...
+      {key:'susceptible', color: colors.susceptible, name:t('Susceptible'), legendType:"line"},
       //{key:'exposed', color: colors.exposed, name:'', legendType:"line"},
-      {key:'infectious', color: colors.infectious, name:'Infectious', legendType:"line"},
+      {key:'infectious', color: colors.infectious, name:t('Infectious'), legendType:"line"},
       // {key:'hospitalized', color: colors.severe, name:'Severely ill', legendType:"line"},
-      {key:'critical', color: colors.critical, name:'Patients in ICU', legendType:"line"},
-      {key:'overflow', color: colors.overflow, name:'ICU overflow', legendType:"line"},
-      {key:'recovered', color: colors.recovered, name:'Recovered', legendType:"line"},
-      {key:'dead', color: colors.death, name:'Cumulative deaths', legendType:"line"},
+      {key:'critical', color: colors.critical, name:t('Patients in icu'), legendType:"line"},
```
Comment (xrmx) [inline src/components/Main/Results/DeterministicLinePlot.tsx]: same

File: src/components/Main/Results/DeterministicLinePlot.tsx [hunk 8]
```diff
@@ -107,16 +110,16 @@ export function DeterministicLinePlot({ data, userResult, logScale, caseCounts }
...
+      {key:'overflow', color: colors.overflow, name:t('Icu overflow'), legendType:"line"},
```
Comment (xrmx) [inline src/components/Main/Results/DeterministicLinePlot.tsx]: same

File: src/components/Main/Results/DeterministicLinePlot.tsx [hunk 9]
```diff
@@ -127,22 +130,22 @@ export function DeterministicLinePlot({ data, userResult, logScale, caseCounts }
...
-        scatterToPlot.push({key:'cases', 'color': colors.cumulativeCases, name: "Cumulative confirmed cases"})
+        scatterToPlot.push({key:'cases', 'color': colors.cumulativeCases, name: t('Cumulative confirmed cases')})
       }
       if (count_observations.hospitalized){
-        scatterToPlot.push({key:'currentHospitalized', 'color': colors.severe, name: "Patients in hospital"})
+        scatterToPlot.push({key:'currentHospitalized', 'color': colors.severe, name: t('Patients in hospital')})
       }
       if (count_observations.ICU){
-        scatterToPlot.push({key:'ICU', 'color': colors.critical, name: "Patients in ICU"})
+        scatterToPlot.push({key:'ICU', 'color': colors.critical, name: t('Patients in icu')})
```
Comment (xrmx) [inline src/components/Main/Results/DeterministicLinePlot.tsx]: same

File: src/components/Main/Results/OutcomeRatesTable.tsx [hunk 10]
```diff
@@ -63,58 +66,58 @@ export function OutcomeRatesTable({ result, rates }: TableProps) {
...
-              <td>Fatal [%]: </td>
+              <td>{t('Fatal')} [%]: </td>
               <td>{deathFrac}</td>
             </tr>
           </tbody>
         </table>
       </Col>
       <Col lg={6}>
-        <h5>Totals/Peak</h5>
+        <h5>{t('Totals')}/{t('Peak')}</h5>
```
Comment (xrmx) [inline src/components/Main/Results/OutcomeRatesTable.tsx]: I think you should keep it as a single string

File: src/components/Main/Results/OutcomeRatesTable.tsx [hunk 11]
```diff
@@ -63,58 +66,58 @@ export function OutcomeRatesTable({ result, rates }: TableProps) {
...
         <table>
           <thead>
             <tr>
-              <th>Quantity &emsp; </th>
-              <th>Peak/total value</th>
+              <th>{t('Quantity')} &emsp; </th>
+              <th>{t('Peak total value')}</th>
```
Comment (xrmx) [inline src/components/Main/Results/OutcomeRatesTable.tsx]: I think you changed both the string and the meaning here

File: src/components/Main/Results/ResultsCard.tsx [hunk 12]
```diff
@@ -52,18 +55,19 @@ function ResultsCard({ canRun, severity, result, caseCounts }: ResutsCardProps)
 
   // TODO: shis should probably go into the `Compare/`
   async function handleFileSubmit(files: Map<FileType, File>) {
+    
     setFiles(files)
 
     const csvFile: File | undefined = files.get(FileType.CSV)
     if (!csvFile) {
-      throw new Error(`Error: CSV file is missing"`)
+      throw new Error(`t('Error'): t('Csv file is missing')`)
```
Comment (xrmx) [inline src/components/Main/Results/ResultsCard.tsx]: CSV

File: src/components/Main/Results/ResultsCard.tsx [hunk 13]
```diff
@@ -52,18 +55,19 @@ function ResultsCard({ canRun, severity, result, caseCounts }: ResutsCardProps)
...
     }
 
     const csvString: string = await readFile(csvFile)
     const { data, errors, meta } = Papa.parse(csvString, { trimHeaders: false })
     if (meta.aborted || errors.length > 0) {
       // TODO: have to report this back to the user
-      throw new Error(`Error: CSV file could not be parsed"`)
+      throw new Error(`t('Error'): t('Csv file could not be parsed')`)
```
Comment (xrmx) [inline src/components/Main/Results/ResultsCard.tsx]: same

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 14]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
   return (
     <CardWithDropdown
       identifier="epidemiologicalScenario"
-      label={<h5 className="p-0 d-inline text-truncate">{'Epidemiology'}</h5>}
-      help="Epidemiological parameters specifing growth rate, seasonal variation, and duration of hospital stay. The presets are combinations of speed and geography (speed/region)."
+      label={<h5 className="p-0 d-inline text-truncate">{t('Epidemiology')}</h5>}
+      help={t('Epidemiological parameters specifing growth rate')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: same

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 15]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
...
       options={epidemiologicalScenarioOptions}
       value={epidemiologicalScenarioOptions.find(s => s.label === scenarioState.epidemiological.current)}
       onValueChange={handleChangeEpidemiologicalScenario}
     >
       <FormSpinBox
         identifier="epidemiological.r0"
-        label={'Annual average R\u2080'}
-        help="Average number of secondary infections per case. When R0 varies throughout the year (seasonal forcing), this value is the mean R0."
+        label={`${t('Annual average')} R\u2080`}
+        help={t('Average number of secondary infections per case')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: same

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 16]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
...
         step={0.1}
         errors={errors}
         touched={touched}
       />
       <FormSpinBox
         identifier="epidemiological.incubationTime"
-        label="Latency [days]"
-        help="Time from infection to onset of symptoms (here onset of infectiousness)"
+        label={`${t('Latency')} [${t('days')}]`}
+        help={t('Time from infection to onset of symptoms here onset of infectiousness')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: same

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 17]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
...
         min={0}
         errors={errors}
         touched={touched}
       />
       <FormSpinBox
         identifier="epidemiological.infectiousPeriod"
-        label="Infectious period [days]"
-        help="Average number of days a person is infectious. Over this time, R0 infections happen on average. Together with the latency time, this defines the serial interval. The longer the serial interval, the slower the outbreak."
+        label={`${t('Infectious period')} [${t('days')}]`}
+        help={t('Average number of days a person is infectious together with the incubation time this defines the serial interval')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: same

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 18]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
...
         step={1}
         min={0}
         errors={errors}
         touched={touched}
       />
       <FormSpinBox
         identifier="epidemiological.lengthICUStay"
-        label="ICU stay [days]"
-        help="Average number of days a critical case stays in the ICU"
+        label={`${t('Icu stay')} [${t('days')}]`}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: ICU

File: src/components/Main/Scenario/ScenarioCardEpidemiological.tsx [hunk 19]
```diff
@@ -37,77 +40,77 @@ function ScenarioCardEpidemiological({
...
+        help={t('Average number of days a critical case stays in the icu')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardEpidemiological.tsx]: same

File: src/components/Main/Scenario/ScenarioCardPopulation.tsx [hunk 20]
```diff
@@ -29,80 +31,81 @@ export interface ScenarioCardPopulationProps {
...
+      help={t('Parameters of the population in the health care system')}
       options={populationScenarioOptions}
       value={populationScenarioOptions.find(s => s.label === scenarioState.population.current)}
       onValueChange={handleChangePopulationScenario}
     >
       <FormSpinBox
         identifier="population.populationServed"
-        label="Population"
-        help="Number of people served by health care system"
+        label={t('PopulationServed')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardPopulation.tsx]: This was Population

File: src/components/Main/Scenario/ScenarioCardPopulation.tsx [hunk 21]
```diff
@@ -29,80 +31,81 @@ export interface ScenarioCardPopulationProps {
...
         step={1}
         errors={errors}
         touched={touched}
       />
       <FormDropdown<string>
         identifier="population.country"
-        label="Age distribution"
-        help="Country to determine the age distribution in the population"
+        label={t('Age distribution')}
+        help={t('Country to determine the Age distribution in the population')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardPopulation.tsx]: age

File: src/components/Main/Scenario/ScenarioCardPopulation.tsx [hunk 22]
```diff
@@ -29,80 +31,81 @@ export interface ScenarioCardPopulationProps {
...
         step={0.1}
         errors={errors}
         touched={touched}
       />
       <FormSpinBox
         identifier="population.hospitalBeds"
-        label="Hospital Beds (est.)"
-        help="Number of hospital beds available in health care system. Presets are rough estimates indicating total capacity. Number of beds available for COVID-19 treatment is likely much lower."
+        label={`${t('Hospital beds')} (${t('est')})`}
+        help={t('Number of hospital beds available in health care system presets are rough estimates indicating total capacity number of beds available for Covid-19 treatment is likely much lower')}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardPopulation.tsx]: you changed the string

File: src/components/Main/Scenario/ScenarioCardPopulation.tsx [hunk 23]
```diff
@@ -29,80 +31,81 @@ export interface ScenarioCardPopulationProps {
...
         step={1}
         errors={errors}
         touched={touched}
       />
       <FormSpinBox
         identifier="population.ICUBeds"
-        label="ICU/ICMU (est.)"
-        help="Number of ICU/ICMUs available in health care system. Presets are rough estimates indicating total capacity. Number of ICU/ICMUs available for COVID-19 treatment is likely much lower."
+        label={`${t('icu-icmu')} (${t('est')})`}
```
Comment (xrmx) [inline src/components/Main/Scenario/ScenarioCardPopulation.tsx]: same

**Comment by hannesgr** [comment]
Yeah, you're right. My assumption was that people will use the english translation.json and translate the values from there. I'll update the texts, thanks!

**Comment by hannesgr** [comment]
Updated now so that keys accurately reflect the translated strings. The edits are in a separate commit for clarity. Just let me know if you'd like me to squash.

**Comment by ivan-aksamentov** [comment]
@hannesgr Fantastic work! We are now going to a planetary scale :)
@xrmx Thanks so much for reviewing!

**Pull request merged**